### PR TITLE
[CFM-888-fix] Block delete group/project through url, for anyone besides site admins.

### PR DESCRIPTION
**Pull request by anvmn**
## Fix for #1172 

https://issuetracker.amplexor.com/jira/browse/CFM-888

### CODE REVIEW

**Comment by anvmn** [comment]
out

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 1]
```diff
@@ -1313,3 +1313,17 @@ function c4m_og_get_user_owned_groups($uid, $node_type) {
...
 }
+
+/**
+ * Implements hook_node_access().
+ *
+ * Prevents anyone besides site administrators to delete Group/Project.
+ * Blocks access to 'node/%node/delete'.
+ */
+function c4m_og_node_access($node, $op, $account) {
+  if ($op == 'delete' && og_is_group('node', $node) && !user_access('administer users')) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: @anvmn 
should be `administer nodes`

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.access.inc [hunk 2]
```diff
@@ -358,8 +358,8 @@ function c4m_og_form_group_node_form_alter(&$form, &$form_state) {
   );
   $form['c4m_related_group'][LANGUAGE_NONE]['#after_build'] = array('c4m_og_select2_widget_after_build');
 
-  // Disable the "delete" option for all users except SA.
-  if (!user_access('administer nodes')) {
+  // Hide "Delete" button for all users except SA.
+  if (!user_access('administer users')) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.access.inc]: should be `administer nodes`

**Comment by ordavidil** [comment]
@anvmn 
Please add 2 behat test under the `admin_access.feature` file.
1. Validate the button exists for SA.
2. Validate the button does not exists for non SA.

**Comment by anvmn** [comment]
@ordavidil 
Is there a way to extend DrupalContext.php?
I need to define a function that locates a button.

File: project/profiles/capacity4more/behat/features/admin_access.feature [hunk 3]
```diff
@@ -20,3 +20,19 @@ Feature: Test custom admin pages
...
+  @api
+  Scenario: Site Admin user visits "Published" group and is allowed to delete it
+    Given I am logged in as user "mariecurie"
+    When  I visit "Published group" node of type "group"
+    Then  I should be allowed to edit a group "Published group"
+    And   I should see the "Delete" button
+    And   I should be allowed to delete a group "Published group"
+
+  @api
+  Scenario: Group Admin user visits "Published" group and is allowed to delete it
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/admin_access.feature]: @anvmn 
`and is not allowed to...`

File: project/profiles/capacity4more/behat/features/bootstrap/FeatureContext/Group.php [hunk 4]
```diff
@@ -698,4 +719,33 @@ public function iResetOrderToAlphabetical() {
     $el = $page->find('xpath', '//a[contains(text(),\'Order items alphabetically\') and not(ancestor::*[contains(@style,\'visibility: hidden\')])]');
     $el->click();
   }
+
+  /**
+   * @Then /^I should see the "([^"]*)" button$/
+   */
+  public function IShouldSeeTheButton($locator) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/bootstrap/FeatureContext/Group.php]: @anvmn 
there should be a function like that alrady

Comment (anvmn) [inline project/profiles/capacity4more/behat/features/bootstrap/FeatureContext/Group.php]: @ordavidil 
Was searching for it, but couldn't find anything like it.
There're functions that search for link /  text, but not for buttons.

File: project/profiles/capacity4more/behat/features/admin_access.feature [hunk 5]
```diff
@@ -20,3 +20,19 @@ Feature: Test custom admin pages
     Given I am logged in as user "mariecurie"
     When  I go to "admin/c4m/groups"
     Then  I should see the groups and the user "mariecurie" in the group management table
+
+  @api @my
```
Comment (ordavidil) [inline project/profiles/capacity4more/behat/features/admin_access.feature]: @anvmn 
great job!

just remove the `@my` tag from both tests

Comment (anvmn) [inline project/profiles/capacity4more/behat/features/admin_access.feature]: @ordavidil 
Done

File: project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module [hunk 6]
```diff
@@ -1657,6 +1657,20 @@ function c4m_og_get_user_owned_groups($uid, $node_type) {
 }
 
 /**
+ * Implements hook_node_access().
+ *
+ * Prevents anyone besides site administrators to delete Group/Project.
+ * Blocks access to 'node/%node/delete'.
+ */
+function c4m_og_node_access($node, $op, $account) {
+  if ($op == 'delete' && og_is_group('node', $node) && !user_access('administer nodes')) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: @anvmn 
did you check it with the user _`mariecurie`_ ?

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/og/c4m_og/c4m_og.module]: just saw the test,  ignore it

**Pull request merged**